docs/tasks/debug-application-cluster/audit.md

```diff
@@ -284,6 +298,138 @@ Events are POSTed as a JSON serialized `EventList`. An example payload:
+              subtype file
+              remove_prefix audit
+              <template>
+                  time_slice_format %Y%m%d%H
+                  compress gz
+                  path /var/log/audit-${tag}.*.log
+                  format json
+                  include_time_key true
+              </template>
+          </match>
+1. start fluentd
+
+          $ fluentd -c /etc/fluentd/config  -vv
+1. start kube-apiserver with the following options:
+
+          --feature-gates=AdvancedAuditing=true --audit-policy-file=/etc/kubernetes/audit-policy.yaml --audit-log-path=/var/log/kube-audit --audit-log-format=json
+1. check audits for different namespaces in /var/log/audit-*.log
+
+#### Use logstash to collect and distribute audit events from webhook backend
+
+[Logstash][logstash] is an open source, server-side data processing tool. In this example,
+we will use logstash to collect audit events from webhook backend, and save events of
+different users into different files.
+
+1. install [logstash][logstash_install_doc]
+1. create config file for logstash
+
+          $ cat <<EOF > /etc/logstash/config
+          input{
+              http{
+                  #TODO, does logstash support https input?
+                  port=>8888
+              }
+          }
+          filter{
+              split{
+                  # Webhook audit backend sends several events together with EventList
+                  # split each event here.
+                  field=>[items]
+                  # We only need event subelement, remove others.
+                  remove_field=>[headers, metadata, apiVersion, "@timestamp", kind, "@version", host]
+              }
+              mutate{
+                  rename => {items=>event}
+              }
+          }
+          output{
+              file{
+                  # Audit events from different users will be saved into different files.
+                  path=>"/var/log/kube-audit-%{[event][user][username]}/audit"
+              }
+          }
+1. start logstash
+
+          $ bin/logstash -f /etc/logstash/config --path.settings /etc/logstash/
+1. create a [kubeconfig file](/docs/tasks/access-application-cluster/authenticate-across-clusters-kubeconfig/) for kube-apiserver webhook audit backend
+
+          $ cat <<EOF > /etc/kubernetes/audit-webhook-kubeconfig
+          apiVersion: v1
+          clusters:
+          - cluster:
+              server: http://<ip_of_logstash>:8888
+            name: logstash
+          contexts:
+          - context:
+              cluster: logstash
+              user: ""
+            name: default-context
+          current-context: default-context
+          kind: Config
+          preferences: {}
+          users: []
+          EOF
+1. start kube-apiserver with the following options:
+
+          --feature-gates=AdvancedAuditing=true --audit-policy-file=/etc/kubernetes/audit-policy.yaml --audit-webhook-config-file=/etc/kubernetes/audit-webhook-kubeconfig
```

Comment:
I guess this feature gate will go away with 1.8. /cc @timstclair @ericchiang we should put that onto the beta TODO list.